From e5293f1c5d7b20d98ed4975dc29a6f88c8bc6a0d Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Thu, 6 Feb 2020 17:55:25 +0100
Subject: [PATCH xserver 04/22] Revert "dri2: Don't make reference to
 noClientException"

It's true that the value would always be -1, if it's not zero, but it's
usually zero is the problem. As a result we return failure from
otherwise successful indirect GLX paths, which isn't very nice of us.

This reverts commit 7d33ab0f8c7958b205076f71e4b47c24aace77fd.

Fixes: https://gitlab.freedesktop.org/xorg/xserver/issues/211
(cherry picked from commit e1fa3beb2fe2519e69f859f0acdc68e5a770de27)
---
 glx/glxdri2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glx/glxdri2.c b/glx/glxdri2.c
index d402ca860..822515a86 100644
--- a/glx/glxdri2.c
+++ b/glx/glxdri2.c
@@ -295,7 +295,7 @@ __glXDRIcontextWait(__GLXcontext * baseContext,
     }
 
     if (ret) {
-        *error = -1;
+        *error = cl->client->noClientException;
         return TRUE;
     }
 
-- 
2.24.1

